Hasher doesn't implement digest::crypto_common::BlockSizeUser

```rs
use hmac::{Mac, SimpleHmac};
use blake3::{Hasher};

fn main() {
   let mut x = SimpleHmac::<Hasher>::new_from_slice(b"test");
   x.update(b"test2");
   println!("{:?}", x.finalize().into_bytes());
}
```

```rs
error[E0599]: the function or associated item `new_from_slice` exists for struct `SimpleHmac<blake3::Hasher>`, but its trait bounds were not satisfied
  --> src/main.rs:5:38
   |
5  |    let mut x = SimpleHmac::<Hasher>::new_from_slice(b"test").unwrap();
   |                                      ^^^^^^^^^^^^^^ function or associated item cannot be called on `SimpleHmac<blake3::Hasher>` due to unsatisfied trait bounds
   |
  ::: /Users/joshuabrown3/.cargo/registry/src/github.com-1ecc6299db9ec823/hmac-0.12.0/src/simple.rs:14:1
   |
14 | pub struct SimpleHmac<D: Digest + BlockSizeUser> {
   | ------------------------------------------------
   | |
   | doesn't satisfy `SimpleHmac<blake3::Hasher>: FixedOutput`
   | doesn't satisfy `SimpleHmac<blake3::Hasher>: KeyInit`
   | doesn't satisfy `SimpleHmac<blake3::Hasher>: MacMarker`
   | doesn't satisfy `SimpleHmac<blake3::Hasher>: Mac`
   | doesn't satisfy `SimpleHmac<blake3::Hasher>: Sized`
   | doesn't satisfy `SimpleHmac<blake3::Hasher>: Update`
   |
   = note: the following trait bounds were not satisfied:
           `SimpleHmac<blake3::Hasher>: Sized`
           which is required by `SimpleHmac<blake3::Hasher>: Mac`
           `SimpleHmac<blake3::Hasher>: KeyInit`
           which is required by `SimpleHmac<blake3::Hasher>: Mac`
           `SimpleHmac<blake3::Hasher>: Update`
           which is required by `SimpleHmac<blake3::Hasher>: Mac`
           `SimpleHmac<blake3::Hasher>: FixedOutput`
           which is required by `SimpleHmac<blake3::Hasher>: Mac`
           `SimpleHmac<blake3::Hasher>: MacMarker`
           which is required by `SimpleHmac<blake3::Hasher>: Mac`
           `&SimpleHmac<blake3::Hasher>: KeyInit`
           which is required by `&SimpleHmac<blake3::Hasher>: Mac`
           `&SimpleHmac<blake3::Hasher>: Update`
           which is required by `&SimpleHmac<blake3::Hasher>: Mac`
           `&SimpleHmac<blake3::Hasher>: FixedOutput`
           which is required by `&SimpleHmac<blake3::Hasher>: Mac`
           `&SimpleHmac<blake3::Hasher>: MacMarker`
           which is required by `&SimpleHmac<blake3::Hasher>: Mac`
           `&mut SimpleHmac<blake3::Hasher>: KeyInit`
           which is required by `&mut SimpleHmac<blake3::Hasher>: Mac`
           `&mut SimpleHmac<blake3::Hasher>: Update`
           which is required by `&mut SimpleHmac<blake3::Hasher>: Mac`
           `&mut SimpleHmac<blake3::Hasher>: FixedOutput`
           which is required by `&mut SimpleHmac<blake3::Hasher>: Mac`
           `&mut SimpleHmac<blake3::Hasher>: MacMarker`
           which is required by `&mut SimpleHmac<blake3::Hasher>: Mac`


error[E0277]: the trait bound `blake3::Hasher: BlockSizeUser` is not satisfied
  --> src/main.rs:5:16
   |
5  |    let mut x = SimpleHmac::<Hasher>::new_from_slice(b"test").unwrap();
   |                ^^^^^^^^^^^^^^^^^^^^ the trait `BlockSizeUser` is not implemented for `blake3::Hasher`
   |
note: required by a bound in `SimpleHmac`
  --> /Users/joshuabrown3/.cargo/registry/src/github.com-1ecc6299db9ec823/hmac-0.12.0/src/simple.rs:14:35
   |
14 | pub struct SimpleHmac<D: Digest + BlockSizeUser> {
   |                                   ^^^^^^^^^^^^^ required by this bound in `SimpleHmac`
```

This is with 

```rs
[dependencies]
hmac = "0.12.0"
blake3 = { version = "1.3.0", features = ["traits-preview"] }
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -93,6 +93,7 @@ page_size = "0.4.1"
 rand = "0.8.0"
 rand_chacha = "0.3.0"
 reference_impl = { path = "./reference_impl" }
+hmac = "0.12.0"
 
 [build-dependencies]
 cc = "1.0.4"

--- a/src/traits.rs
+++ b/src/traits.rs
@@ -170,4 +170,13 @@ mod test {
         let out2 = digest::Mac::finalize(hasher2);
         assert_eq!(out1.as_bytes(), out2.into_bytes().as_slice());
     }
+
+    #[test]
+    fn test_hmac_compatibility() {
+        use hmac::{SimpleHmac, Mac};
+
+        let mut x = SimpleHmac::<Hasher>::new_from_slice(b"key").unwrap();
+        hmac::digest::Update::update(&mut x, b"data");
+        assert_ne!(x.finalize().into_bytes().len(), 0);
+    }
 }

--- a/src/traits.rs
+++ b/src/traits.rs
@@ -5,7 +5,7 @@ pub use digest;
 
 use crate::{Hasher, OutputReader};
 use digest::crypto_common;
-use digest::generic_array::{typenum::U32, GenericArray};
+use digest::generic_array::{typenum::U32, typenum::U64, GenericArray};
 
 impl digest::HashMarker for Hasher {}
 
@@ -69,6 +69,10 @@ impl digest::XofReader for OutputReader {
 
 impl crypto_common::KeySizeUser for Hasher {
     type KeySize = U32;
+}
+
+impl crypto_common::BlockSizeUser for Hasher {
+    type BlockSize = U64;
 }
 
 impl digest::MacMarker for Hasher {}

